tailscale/resource_dns_split_nameservers.go

```diff
@@ -0,0 +1,112 @@
+package tailscale
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/tailscale/tailscale-client-go/tailscale"
+)
+
+func resourceDNSSplitNameservers() *schema.Resource {
+	return &schema.Resource{
+		Description:   "The dns_split_nameservers resource allows you to configure split DNS nameservers for your Tailscale network. See https://tailscale.com/kb/1054/dns for more information.",
+		ReadContext:   resourceSplitDNSNameserversRead,
+		CreateContext: resourceSplitDNSNameserversCreate,
+		UpdateContext: resourceSplitDNSNameserversUpdate,
+		DeleteContext: resourceSplitDNSNameserversDelete,
+		Schema: map[string]*schema.Schema{
+			"domain": {
+				Type:        schema.TypeString,
+				Description: "Domain to configure split DNS for. Requests for this domain will be resolved using the provided nameservers.",
+				Required:    true,
+			},
+			"nameservers": {
+				Type:        schema.TypeSet,
+				Description: "Devices on your network will use these nameservers to resolve DNS names. IPv4 or IPv6 addresses are accepted.",
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func resourceSplitDNSNameserversRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*tailscale.Client)
+	splitDNS, err := client.DNSSplitDNS(ctx)
+	if err != nil {
+		return diagnosticsError(err, "Failed to fetch split DNS configs")
+	}
+
+	nameservers := splitDNS[d.Get("domain").(string)]
```

Comment:
We might need something like this to consider the resource deleted when there are no split DNS servers configured for a given domain.

```suggestion
	nameservers, ok := splitDNS[d.Get("domain").(string)]
	if !ok {
	  d.SetId("")
	  return
	}
```

Reply:
Digging into this a bit more but making this change was breaking import.

Reply:
Ended up changing to use `d.Id()` as this is what was breaking the import and is mentioned as necessary [here](https://developer.hashicorp.com/terraform/plugin/sdkv2/resources/import#importer-state-function) if using `schema.ImportStatePassthrough`.

This seems to operate fine without the `ok` check in the scenarios I've tested (namely creation, deletion, import, update of split DNS nameservers) so I've left it out for now.